benchmark/spmv/batch_spmv.cpp

```diff
@@ -0,0 +1,484 @@
+        std::vector<gko::size_type> tuning_values{
+            1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 2048, 4096};
+        for (auto val : tuning_values) {
+            // Actually set the value that will be tuned. See
+            // cuda/components/format_conversion.cuh for an example of how this
+            // variable is used.
+            gko::_tuned_value = val;
+            auto tuning_timer = get_timer(exec, FLAGS_gpu_timer);
+            for (unsigned int i = 0; i < FLAGS_repetitions; i++) {
+                auto x_clone = clone(x);
+                exec->synchronize();
+                tuning_timer->tic();
+                system_matrix->apply(lend(b), lend(x_clone));
+                tuning_timer->toc();
+            }
+            tuning_case["time"].PushBack(tuning_timer->compute_average_time(),
+                                         allocator);
+            tuning_case["values"].PushBack(val, allocator);
+        }
+        // We put back the flag to false to use the default (non-tuned) values
+        // for the following
+        gko::_tuning_flag = false;
+#endif  // GINKGO_BENCHMARK_ENABLE_TUNING
+
+        // timed run
+        auto timer = get_timer(exec, FLAGS_gpu_timer);
+        for (unsigned int i = 0; i < FLAGS_repetitions; i++) {
+            auto x_clone = clone(x);
+            exec->synchronize();
+            timer->tic();
+            system_matrix->apply(lend(b), lend(x_clone));
+            timer->toc();
+        }
+        add_or_set_member(spmv_case[format_name], "num_batch_entries",
+                          system_matrix->get_num_batch_entries(), allocator);
+        add_or_set_member(spmv_case[format_name], "time",
+                          timer->compute_average_time(), allocator);
+
+        // compute and write benchmark data
+        add_or_set_member(spmv_case[format_name], "completed", true, allocator);
+    } catch (const std::exception &e) {
+        add_or_set_member(test_case["spmv"][format_name], "completed", false,
+                          allocator);
+        std::cerr << "Error when processing test case " << test_case << "\n"
+                  << "what(): " << e.what() << std::endl;
+    }
+}
+
+// This function supposes that management of `FLAGS_overwrite` is done before
+// calling it
+//
+// Overload with the reading a batch matrix with std::vector<matrix_data>
+void apply_spmv(const char *format_name, std::shared_ptr<gko::Executor> exec,
+                const std::vector<gko::matrix_data<etype>> &data,
+                const batch_vec<etype> *b, const batch_vec<etype> *x,
+                const batch_vec<etype> *answer, rapidjson::Value &test_case,
+                rapidjson::MemoryPoolAllocator<> &allocator)
+{
+    try {
+        auto &spmv_case = test_case["spmv"];
+        add_or_set_member(spmv_case, format_name,
+                          rapidjson::Value(rapidjson::kObjectType), allocator);
+
+        auto nbatch = FLAGS_num_duplications;
```

Comment:
Just for clarity, as this is no longer the number of matrices in the batch.
```suggestion
        auto n_dup = FLAGS_num_duplications;
```